### ISSUE TITLE
Duplicate email usage

### ISSUE THREAD

**Issue by NishantPacharne** [pennersr/django-allauth#3076]
**The Bug**

- If a user signs up from our signup form and uses abc@gmail.com as mail id there.
- Still that mail id can be used to sign up if continuing with google
- Yes after continuing with google, a form is supposed to be filled with email and username, in this form the duplicate usage of email is validated and user is warned. but then a different mail id (xyz@gmail.com) can be temporarily provided in the form and the sign up gets completed.
- So basically what happens is that, the user having email xyz@gmail.com, his user accounts gets linked to google account (abc@gmail.com)
- Now if you initiate a password reset request for abc@gmail.com, then 2 password reset mails are sent to abc@gmail.com, one is for himself and one is for xyz@gmail.com.

**Comment by esalitrynskiy** [comment]
Any fix for this?

**Comment by pennersr** [comment]
See #3374 -- `SOCIALACCOUNT_EMAIL_AUTHENTICATION = True` will result in the user signing in into the already existing local account.

**Comment by pennersr** [comment]
> Now if you initiate a password reset request for [abc@gmail.com](mailto:abc@gmail.com), then 2 password reset mails are sent to [abc@gmail.com](mailto:abc@gmail.com), one is for himself and one is for [xyz@gmail.com](mailto:xyz@gmail.com).

Fixed via f98c9a3b

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3076
